lib/msal-node/src/utils/RetryTracker.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Logger } from "@azure/msal-common";
+
+export class RetryTracker {
+    private logger: Logger;
+    private retryLimit: number;
+    private retryWaitTimeMilli: number;
+
+    private _numRetried: number;
+    public resetNumRetried(): void {
+        this._numRetried = 0;
+    }
+
+    constructor(
+        logger: Logger,
+        retryLimit: number,
+        retryWaitTimeMilli: number
+    ) {
+        this.logger = logger;
+        this.retryLimit = retryLimit;
+        this.retryWaitTimeMilli = retryWaitTimeMilli;
+        this._numRetried = 0;
+    }
+
+    // HeaderNames.RETRY_AFTER
+    public async retry(
+        statusCode: number,
+        retryAfterHeader: string | undefined
+    ): Promise<boolean> {
+        if (this._numRetried === this.retryLimit) {
+            this.logger.warning(
+                "The request can not be retried because it has already been retried the maximum number of times."
+            );
+            return false;
+        }
+
+        if (!retryAfterHeader) {
```

Comment:
@neha-bhargava @Avery-Dunn  - what are we doing in MSALs when IMDS returns a RetryAfter header? IMO we should respect this header, but still retry.

Reply:
@bgavrilMS I don't think IMDS or any other managed identity endpoints send back a retry-after header.